service/hub/api/authn.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+import "errors"
+
+type Labels map[string][]string
+
+// Authenticator Authentication plugin interface.
+type Authenticator interface {
+	// Authenticate Given a username and a password (plain text), responds with the result or an error.
+	// Error should only be reported if request could not be serviced, not if it should be denied.
+	// A special NoMatch error is returned if the authorizer could not reach a decision,
+	// e.g. none of the rules matched.
+	// Another special WrongPass error is returned if the authorizer failed to authenticate.
+	// Implementations must be goroutine-safe.
+	Authenticate(user string, password PasswordString) (bool, Labels, error)
```

Comment:
What the labels for ?

Reply:
labels can carry variables from authn to authz, so it can be more convenient to identify user/org.